hapi-fees/src/main/java/com/hedera/services/usage/token/TokenMintUsage.java

```diff
@@ -41,9 +45,26 @@ TokenMintUsage self() {
 		return this;
 	}
 
+
+	public TokenMintUsage givenSubType(SubType subType) {
+		this.currentSubType = subType;
+		return this;
+	}
+
 	public FeeData get() {
+		var op = this.op.getTokenMint();
+
+		// TODO: Split logic for calculations
+		if (currentSubType == SubType.TOKEN_NON_FUNGIBLE_UNIQUE) {
```

Comment:
Can be done in a future PR, but we've noticed that the Streams API has too much overhead for use in the `handleTransaction` critical perf path.

Reply:
I replaced the logic accordingly.